src/StoreApi/Routes/V1/CheckoutOrder.php

```diff
@@ -0,0 +1,432 @@
+	 *
+	 * @var \WC_Order
+	 */
+	private $order = null;
+
+	/**
+	 * Get the path of this REST route.
+	 *
+	 * @return string
+	 */
+	public function get_path() {
+		return '/checkout/(?P<id>[\d]+)';
+	}
+
+	/**
+	 * Checks if a nonce is required for the route.
+	 *
+	 * @param \WP_REST_Request $request Request.
+	 * @return bool
+	 */
+	protected function requires_nonce( \WP_REST_Request $request ) {
+		return true;
+	}
+
+	/**
+	 * Get method arguments for this REST route.
+	 *
+	 * @return array An array of endpoints.
+	 */
+	public function get_args() {
+		return [
+			[
+				'methods'             => \WP_REST_Server::CREATABLE,
+				'callback'            => [ $this, 'get_response' ],
+				'permission_callback' => [ $this, 'is_authorized' ],
+				'args'                => array_merge(
+					[
+						'payment_data' => [
+							'description' => __( 'Data to pass through to the payment method when processing payment.', 'woo-gutenberg-products-block' ),
+							'type'        => 'array',
+							'items'       => [
+								'type'       => 'object',
+								'properties' => [
+									'key'   => [
+										'type' => 'string',
+									],
+									'value' => [
+										'type' => [ 'string', 'boolean' ],
+									],
+								],
+							],
+						],
+					],
+					$this->schema->get_endpoint_args_for_item_schema( \WP_REST_Server::CREATABLE )
+				),
+			],
+			'schema'      => [ $this->schema, 'get_public_item_schema' ],
+			'allow_batch' => [ 'v1' => true ],
+		];
+	}
+
+	/**
+	 * Prepare a single item for response. Handles setting the status based on the payment result.
+	 *
+	 * @param mixed            $item Item to format to schema.
+	 * @param \WP_REST_Request $request Request object.
+	 * @return \WP_REST_Response $response Response data.
+	 */
+	public function prepare_item_for_response( $item, \WP_REST_Request $request ) {
+		$response     = parent::prepare_item_for_response( $item, $request );
+		$status_codes = [
+			'success' => 200,
+			'pending' => 202,
+			'failure' => 400,
+			'error'   => 500,
+		];
+
+		if ( isset( $item->payment_result ) && $item->payment_result instanceof PaymentResult ) {
+			$response->set_status( $status_codes[ $item->payment_result->status ] ?? 200 );
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Process an order.
+	 *
+	 * 1. Process Request
+	 * 2. Process Customer
+	 * 3. Validate Order
+	 * 4. Process Payment
+	 *
+	 * @throws RouteException On error.
+	 * @throws InvalidStockLevelsInCartException On error.
+	 *
+	 * @param \WP_REST_Request $request Request object.
+	 *
+	 * @return \WP_REST_Response
+	 */
+	protected function get_route_post_response( \WP_REST_Request $request ) {
+		$order_id    = absint( $request['id'] );
+		$this->order = wc_get_order( $order_id );
+
+		if ( $this->order->get_status() !== 'pending' ) {
+			return new \WP_Error(
+				'invalid_order_update_status',
+				__( 'This order is not pending and cannot be paid for.', 'woo-gutenberg-products-block' )
```

Comment:
I'd just simplify to:

```suggestion
				__( 'This order cannot be paid for', 'woo-gutenberg-products-block' )
```

Reply:
Updated here [138fcbf](https://github.com/woocommerce/woocommerce-blocks/pull/10287/commits/138fcbf9ebe6c4bc89fcb416df101f4330c81a82)